### ISSUE TITLE
Siblings in directory dependency graph are not drawn in same parent directory.

### ISSUE THREAD

**Issue by dhebbeker** [doxygen/doxygen#8293]
Description
=========

In directory dependency graphs for directories, which have a dependency to a sibling directory, the sibling is not drawn in the parent directory. A sibling directory is a directory which has the same parent directory.

See for example the following directory structure (truncated):

```
src/
├── a
│   ├── aa
│   └── depends_on_b.h
├── b
│   ├── bb
│   ├── dependent.h
│   └── depends_on_cc.h
├── c
│   ├── cc
│   └── dependent.h
└── d
    └── dd
```

In directory dependency graph for `a`, `b` is drawn outside of `src`.

![dir_85293cd6fb67eb83825626cd83b71cea_dep_MASTER](https://user-images.githubusercontent.com/1594340/103443176-ccb44300-4c5c-11eb-97a3-bee30ad732ac.png)

The directory dependency graph is on the Directory Reference page for directory `a`: File --> select directory `a`.

To Reproduce
-------------------

[move-parent-directory-example.zip](https://github.com/doxygen/doxygen/files/5759190/move-parent-directory-example.zip)

Version
----------

Version self-compiled: 1.9.1 (bde178069e1730022e2bde0f083b7e9a6e35fedd)

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
```

Expected behavior
==============

For example in the directory dependency graph for `a` the graph should be drawn such as `a` and its dependee `b` are inside of `src`.

![dir_85293cd6fb67eb83825626cd83b71cea_dep](https://user-images.githubusercontent.com/1594340/103443196-f40b1000-4c5c-11eb-82bc-c4ce44dd0113.png)

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of an official release).
Please don't use the feature "Linked pull requests" as this might disrupt the administration process.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.1.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
